pkg/controller/master/setting/cluster_registration_url.go

```diff
@@ -0,0 +1,67 @@
+package setting
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/rancher/wrangler/pkg/objectset"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/yaml"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+)
+
+// registerCluster imports Harvester to Rancher by applying manifests from the registration URL.
+func (h *Handler) registerCluster(setting *harvesterv1.Setting) error {
```

Comment:
if the user deletes the imported harvester cluster on the Rancher side, we should not re-import the cluster again after the Harvester API server reboot.

And I was wondering should we can reset this setting from the upstream Rancher or deal it on the downstream Harvester?

Reply:
When users delete it upstream, it will not be re-imported after Harvester restarts. We have the setting hash so that it is done once.